File: JavaScript/프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
    var bridge = Array(bridge_length).fill(0);
    var time = 0;
    var rest_weight = weight;
    
    while (truck_weights.length > 0) {
        time += 1;
        
        rest_weight -= bridge[bridge.length - 1];
        rest_weight += bridge[0];
        
        bridge.shift();
        
        let temp = 0;
        if (rest_weight >= truck_weights[0]) {
            temp = truck_weights.shift();
        }
        bridge.push(temp);   
    }
    return time + bridge_length;
}

File: Java/프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.java
import java.util.*;

class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
        int time = 0;
        int rest_weight = weight;
        List<Integer> bridge = new ArrayList<>();
        List<Integer> trucks = new ArrayList<>();
        for (int i=0; i<bridge_length; i++) bridge.add(0);
        for (int j=0; j<truck_weights.length; j++) trucks.add(truck_weights[j]);
        
        while(trucks.size() > 0) {
            time += 1;
            
            rest_weight -= bridge.get(bridge.size() - 1);
            rest_weight += bridge.get(0);
            
            bridge.remove(0);
            int temp = 0;
            if (rest_weight >= trucks.get(0)) temp = trucks.remove(0);
            bridge.add(temp);
        }
        return time + bridge_length;
    }
}   